src/gpu/nvidia/sycl_cuda_scoped_context.cpp

```diff
@@ -26,8 +26,9 @@ cuda_sycl_scoped_context_handler_t::cuda_sycl_scoped_context_handler_t(
         const sycl_cuda_engine_t &engine)
     : need_to_recover_(false) {
     try {
-        auto desired = engine.get_underlying_context();
         CUDA_EXECUTE_FUNC(cuCtxGetCurrent, &original_);
+        auto desired = engine.get_underlying_context();
```

Comment:
Is any change required in the code below? https://github.com/oneapi-src/oneDNN/blob/18932605309f43966bf7b01ce4b279cb2ac77c68/src/gpu/nvidia/sycl_cuda_engine.cpp#L140

It looks like it was implemented based on the assumption that the `CUcontext` is queried from SYCL context and it can be different from the primary context.

Another question, there is a comment that the retain and release functions are expensive (~75ms). Is it still the case? If it is then it looks like the change ins this PR will add that overhead on top of the execution time.

Reply:
I didn't spot this other file, I will change this as well. 

As far as performance goes, the [Nvidia docs](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__DRIVER.html) say that perf will be degraded if using multiple contexts per device. I figure that the comment is related to this, since we were potentially using multiple `CUcontexts` per device before we started using the primary context in DPC++. In my own limited testing there is very little overhead (1-2ms) setting a primary context. `cuDevicePrimaryContextRetain` is not listed in an nsys trace so maybe it is a no op.

Reply:
Yes sorry changes to come. I misread your original comment